compiler/src/iree/compiler/Codegen/Common/FastMathPatterns.cpp

```diff
@@ -19,109 +21,116 @@ namespace {
 struct FastErfPattern : public OpRewritePattern<math::ErfOp> {
   using OpRewritePattern::OpRewritePattern;
 
+  // Helper to evaluate a polynomial using FMA chains for both scalar and vector
+  // types.
+  template <typename ValueT>
+  static ValueT evalErfPolynomial(ValueT ax, ValueT t, ArrayRef<ValueT> coeffs,
+                                  PatternRewriter &rewriter, Location loc) {
+    ValueT acc = coeffs[0];
+    for (size_t i = 1; i < coeffs.size(); ++i) {
+      acc = rewriter.create<math::FmaOp>(loc, acc, t, coeffs[i]);
+    }
+    return rewriter.create<math::FmaOp>(loc, ax, acc, ax);
+  }
+
   LogicalResult matchAndRewrite(math::ErfOp op,
                                 PatternRewriter &rewriter) const override {
     Location loc = op.getLoc();
     Value input = op.getOperand();
     Type resultType = op.getType();
 
-    // Erf only supports f32.
-    if (!resultType.isF32()) {
-      return rewriter.notifyMatchFailure(op, "Result only supports f32");
+    // Only support f32 and vector<f32> types.
+    if (!(resultType.isF32() ||
+          (llvm::isa<VectorType>(resultType) &&
+           llvm::cast<VectorType>(resultType).getElementType().isF32()))) {
+      return rewriter.notifyMatchFailure(
+          op, "Result only supports f32 or vector<f32>");
     }
 
-    // Create constants.
-    Type f32Type = rewriter.getF32Type();
-    auto oneF = arith::ConstantOp::create(rewriter, loc, f32Type,
-                                          rewriter.getF32FloatAttr(1.0f));
-
-    // Get abs value.
-    Value ax = math::AbsFOp::create(rewriter, loc, input);
-
-    // Create comparison for |x| < 1.0.
-    Value cmp = arith::CmpFOp::create(rewriter, loc, arith::CmpFPredicate::OLT,
-                                      ax, oneF);
-
-    // Create if statement.
-    auto ifOp = scf::IfOp::create(rewriter, loc, resultType, cmp, true);
-
-    // --- Then region (|x| < 1.0) ---
-    {
-      OpBuilder::InsertionGuard guard(rewriter);
+    // Helper to create constants for both scalar and vector types.
+    auto createConst = [&](float v) -> Value {
+      if (auto vecType = llvm::dyn_cast<VectorType>(resultType)) {
+        SmallVector<float> values(vecType.getNumElements(), v);
+        return rewriter.create<arith::ConstantOp>(
+            loc, vecType,
+            DenseElementsAttr::get(vecType, ArrayRef<float>(values)));
```

Comment:
nit: you don't need the cast. It is common to not have a cast from SmallVector to ArrayRef